### ISSUE TITLE
Questions about final tokens in hyper-parameter settings

### ISSUE THREAD

**Issue by tsunehiko** [elicassion/StARformer#6]
Hello, thank you for your wonderful work. I have a question regarding the setting of hyperparameters. In the hyper-parameter settings for Atari, what does "Final tokens" refer to? It is mentioned as 2 x 500000 x K, but what does the 2 represent? Additionally, the original Decision Transformer paper includes a mention of Max Epoch, but what about in StARformer? I would be delighted if you could respond to this!

**Comment by elicassion** [comment]
Hi @Tsunehiko,

Thanks for your interest! The "final token" is a setting about training scheduling inherited from Decision Transformer. 2 means there are two kinds of tokens in the Sequence Transformer (StAR-representation and state). Emprically you can set it to other values according to any advanced scheduling strategy for training a Transformer (or Decision Transformer).

The maximum epoch I used to train StARformer is 15. Since StARformer builds on top of Decision Transformer, I observe the same instability in training that the model's performance varies drastically across epochs. Because this is offline learning, I would recommend that using early stopping to get the checkpoint with best score, and then evaluating the final performance with different (and more) seeds.

**Comment by tsunehiko** [comment]
Thank you for your quick response! I understand that in the Sequence Transformer of StARFormer, 2 refers to the types of tokens. On the other hand, the original Decision Transformer also uses 2, but it has three types of tokens: return-to-go, state, and action. Do you know anything about this difference?

Thank you also for the information about the maximum epoch. For the evaluation of the model for early stopping, would you use the scores from several episodes in the actual environment with the trained model?

**Comment by elicassion** [comment]
Sorry for the confusion. I realized that 2 means the number of epochs!

Yes we need to use them.

**Comment by tsunehiko** [comment]
I'm still confused. In the case of StARformer, does the 2 refer to the epoch chosen as a result of the early stopping?

**Comment by elicassion** [comment]
No, the final token is not related to stopping. It refers to reducing
learning rate after “final token” number of tokens.

**Comment by tsunehiko** [comment]
I understand now! Thank you for answering my questions!!
